selfdrive/ui/ui.cc

```diff
@@ -387,207 +364,146 @@ static void update_status(UIState *s, int status) {
-    s->scene.engaged = datad.enabled;
-    s->scene.engageable = datad.engageable;
-    s->scene.gps_planner_active = datad.gpsPlannerActive;
-    s->scene.monitoring_active = datad.driverMonitoringOn;
-
-    s->scene.decel_for_model = datad.decelForModel;
-
-    if (datad.alertSound != cereal_CarControl_HUDControl_AudibleAlert_none && datad.alertSound != s->alert_sound) {
-      if (s->alert_sound != cereal_CarControl_HUDControl_AudibleAlert_none) {
+    scene.v_cruise = data.getVCruise();
+    scene.v_ego = data.getVEgo();
+    scene.curvature = data.getCurvature();
+    scene.engaged = data.getEnabled();
+    scene.engageable = data.getEngageable();
+    scene.gps_planner_active = data.getGpsPlannerActive();
+    scene.monitoring_active = data.getDriverMonitoringOn();
+
+    scene.decel_for_model = data.getDecelForModel();
+    auto alert_sound = data.getAlertSound();
+    const auto sound_none = cereal::CarControl::HUDControl::AudibleAlert::NONE;
+    if (alert_sound != s->alert_sound){
+      if (s->alert_sound != sound_none){
         stop_alert_sound(s->alert_sound);
       }
-      play_alert_sound(datad.alertSound);
-
-      s->alert_sound = datad.alertSound;
-      snprintf(s->alert_type, sizeof(s->alert_type), "%s", datad.alertType.str);
-    } else if ((!datad.alertSound || datad.alertSound == cereal_CarControl_HUDControl_AudibleAlert_none)
-                  && s->alert_sound != cereal_CarControl_HUDControl_AudibleAlert_none) {
-      stop_alert_sound(s->alert_sound);
-      s->alert_type[0] = '\0';
-      s->alert_sound = cereal_CarControl_HUDControl_AudibleAlert_none;
-    }
-
-    if (datad.alertText1.str) {
-      snprintf(s->scene.alert_text1, sizeof(s->scene.alert_text1), "%s", datad.alertText1.str);
-    } else {
-      s->scene.alert_text1[0] = '\0';
-    }
-    if (datad.alertText2.str) {
-      snprintf(s->scene.alert_text2, sizeof(s->scene.alert_text2), "%s", datad.alertText2.str);
-    } else {
-      s->scene.alert_text2[0] = '\0';
-    }
-
-    s->scene.alert_ts = eventd.logMonoTime;
-
-    s->scene.alert_size = datad.alertSize;
-    if (datad.alertSize == cereal_ControlsState_AlertSize_none) {
-      s->alert_size = ALERTSIZE_NONE;
-    } else if (datad.alertSize == cereal_ControlsState_AlertSize_small) {
-      s->alert_size = ALERTSIZE_SMALL;
-    } else if (datad.alertSize == cereal_ControlsState_AlertSize_mid) {
-      s->alert_size = ALERTSIZE_MID;
-    } else if (datad.alertSize == cereal_ControlsState_AlertSize_full) {
-      s->alert_size = ALERTSIZE_FULL;
+      if (alert_sound != sound_none){
+        play_alert_sound(alert_sound);
+        s->alert_type = data.getAlertType();
+      }
+      s->alert_sound = alert_sound;
     }
-
-    if (datad.alertStatus == cereal_ControlsState_AlertStatus_userPrompt) {
+    scene.alert_text1 = data.getAlertText1();
+    scene.alert_text2 = data.getAlertText2();
+    scene.alert_ts = event.getLogMonoTime();
+    scene.alert_size = (uint8_t)data.getAlertSize();
```

Comment:
Don't cast this, it's making an assumption of the numbers of the capnp enum. Instead, we should remove the ALERTSIZE_NONE... defines and use the cereal ones everywhere.